src/payers/hypha_payer.cpp

```diff
@@ -18,9 +20,19 @@ namespace hypha
         TRACE_FUNCTION()
         issueToken(m_dao.getSettingOrFail<eosio::name>(HYPHA_TOKEN_CONTRACT),
                    m_dao.get_self(),
-                   recipient,
+                   m_dao.getSettingOrFail<eosio::name>(HYPHA_COSALE_CONTRACT),
                    quantity,
-                   memo);
+                   "");
+
+        eosio::action(
+            eosio::permission_level{m_dao.get_self(), name("active")},
+            m_dao.getSettingOrFail<eosio::name>(HYPHA_COSALE_CONTRACT), name("createlock"),
+            std::make_tuple(m_dao.get_self(),
+                            recipient,
+                            quantity,
+                            memo,
+                            eosio::time_point(eosio::current_time_point().time_since_epoch() + EXPIRY_MILLISECONDS)))
+            .send();
```

Comment:
time_since_epoch() returns in microseconds, although we are adding to milliseconds (EXPIRY_MILLISECONDS). We should convert to EXPIRY_MILLISECONDS to microseconds as well to get the right amount.

Reply:
Oh gosh, I confused the two. Thanks!